src/Mod/Spreadsheet/App/Cell.cpp

```diff
@@ -279,12 +280,12 @@ void Cell::afterRestore() {
 void Cell::setContent(const char * value)
 {
     PropertySheet::AtomicPropertyChange signaller(*owner);
-    App::Expression * expr = 0;
+    App::Expression * expr = nullptr;
```

Comment:
Can this be changed this to an owning smart pointer that then gets moved at the end? This is pretty complex code and making sure every single path results in a deallocation is asking for trouble. (I understand if you don't want to do it in this PR, but I'm thinking since you're already touching so much of the function, that bit of refactoring might still be a good idea).

Reply:
Speaking of "asking for trouble" -- a member variable called "expression" and a local called "expr" used in the same function is pretty confusing. (Again, I recognize that's not your doing... just that this is an opportunity to clean it up).

Reply:
Maybe we can add a comment in the code for these 2 things @chennes so it can be done by someone else at a later time. But if you can elaborate a bit on the smart pointer I can try too

Reply:
I think a `unique_ptr` will work here, and then replace all of the new statements with `make_unique` (in keeping with the Core Guidelines). Lines 327 and 351 will give you a problem, but those are the memory leaks that this sort of change is designed to prevent. I don't know what an `App::ExpressionPtr` is or how its constructor works, so it may be that it should be used instead of `unique_ptr`, and I also don't know exactly what line 377 needs to be -- some kind of move, but it may be that using an `App::ExpressionPtr` all the way through would take care of that.

Reply:
@chennes: `ExpressionPtr` is a `unique_ptr` (`typedef std::unique_ptr<Expression> ExpressionPtr;`).
I've converted it to use `unique_ptr`, `make_unique` and `move` as suggested. It is probably safer this way, but with the downside of being a bit more verbose. A price worth paying if you ask me :)

I've also renamed `expr`to `newExpr` to avoid mixup with the member variable `expression`.

Please check the latest commit and let me know if you want me to fix anything more.

Reply:
The latest commit compiles and runs fine for me on Windows, and the changes are exactly what I was looking for -- thanks for taking the time to incorporate it into this PR. The Travis Linux builds are currently failing, I believe due to a missing `#include <memory>`.

Reply:
Oh crap I totally forgot that `make_unique` was first introduced in c++14, IIRC FreeCAD is only using c++11.
There's a compatibility layer which can be used instead here: https://github.com/FreeCAD/FreeCAD/blob/ab4ab444b66f675a2c8be39b41ad318f10998d82/src/Base/StdStlTools.h

I'll add this and check if travis can compile afterwards (I cannot test locally as I have to use c++14 due to boost v1.75.0)

Reply:
I've made the changes now, hopefully it compiles this time 🤞